addons/product/models/product.py

```diff
@@ -263,9 +263,28 @@ def _set_product_lst_price(self):
             value -= product.price_extra
             product.write({'list_price': value})
 
+    @api.depends_context('pricelist')
     def _compute_product_price_extra(self):
         for product in self:
-            product.price_extra = sum(product.product_template_attribute_value_ids.mapped('price_extra'))
+            price_extra = sum(product.product_template_attribute_value_ids.mapped('price_extra'))
+            pricelist_id_or_name = self._context.get('pricelist')
+            if pricelist_id_or_name:
+                pricelist = None
+
+                # Support context pricelists specified as list, display_name or ID for compatibility
+                if isinstance(pricelist_id_or_name, list):
+                    pricelist_id_or_name = pricelist_id_or_name[0]
+                if isinstance(pricelist_id_or_name, str):
+                    pricelist_name_search = self.env['product.pricelist'].name_search(pricelist_id_or_name, operator='=', limit=1)
+                    if pricelist_name_search:
+                        pricelist = self.env['product.pricelist'].browse([pricelist_name_search[0][0]])
+                elif isinstance(pricelist_id_or_name, int):
+                    pricelist = self.env['product.pricelist'].browse(pricelist_id_or_name)
```

Comment:
Can't we make a private util method for this? This is exactly the same code as in `_compute_product_price()`.
Will be easier to read those methods, and easier to get rid in master (prior?).

Reply:
Same here :+1: